components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1391,12 +1391,30 @@ where
                 }
             }
             SignificantMsg::StoreUnreachable { store_id } => {
-                if let Some(peer_id) = find_peer(self.region(), store_id).map(|p| p.get_id()) {
+                if let Some(peer) = find_peer(self.region(), store_id).cloned() {
+                    let peer_id = peer.get_id();
                     if self.fsm.peer.is_leader() {
                         self.fsm.peer.raft_group.report_unreachable(peer_id);
+                        let timeout = self.fsm.peer.raft_group.raft.randomized_election_timeout();
+                        self.fsm.peer.countdown_before_hibernate = timeout;
```

Comment:
Doesn't seem necessary. It should be woken up by follower instead. Otherwise, a down peer will wake up all the leaders and break hibernate regions.